library.js

```diff
@@ -69,6 +69,25 @@ plugin.init = function (params, callback) {
 	router.get('/api/admin/plugins/session-sharing', controllers.renderAdminPage);
 
 	router.get('/api/session-sharing/lookup', controllers.retrieveUser);
+	router.post('/api/session-sharing/user', (req, res) => {
+		async.waterfall([
```

Comment:
By removing the `jwt.verify` step here, you're removing the only security check, meaning someone could look at the source and use this API to create as many users as they want (admins too!).

I'd suggest adding the jwt.verify line back, and sending a JWT in the post body instead.

Then `plugin.process` can be re-used in its entirety and called from within the controller.

Reply:
Thats a very good point! Will update the PR later today with changes to how you suggest